File: code_eval/cpp/easy/challenge_01.cpp
// Created by cschalk on 3/21/16.
// Challenge 01 : Fizz Buzz
#include <iostream>
#include <fstream>

using namespace std;

void FizzBuzz(int divisor1, int divisor2, int digits) {
    for (int count = 1 ; count <= digits ; count++) {
        if (count % divisor1 == 0) {
            std::cout << 'F';
            if (count % divisor2 == 0) {
                std::cout << 'B';
            }
        } else if (count % divisor2 == 0) {
            std::cout << 'B';
        } else {
            std::cout << count;
        }
        std::cout << ' ';
    }
    std::cout << std::endl;
}

int main(int argc, char ** argv) {

    std::ifstream input {argv[1]};
    int div1, div2, digits;

    // firstLine
    input >> div1 >> div2 >> digits;
    while (input) {
        FizzBuzz(div1, div2, digits);
        input >> div1 >> div2 >> digits;
    }
    return 0;
}

File: code_eval/java/easy/Challenge01.java
package easy;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 01 : FizzBuzz
 * Created by cschalk on 3/21/16.
 */
public class Challenge01 {

    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            playFizzBuzz(line);
        }
    }

    private static void playFizzBuzz(String input) {
        String[] values = input.split(" ");
        int divisor1 = Integer.parseInt(values[0]);
        int divisor2 = Integer.parseInt(values[1]);
        int limit = Integer.parseInt(values[2]);

        StringBuilder out = new StringBuilder();
        for (int i = 1 ; i <= limit ; i++) {
            boolean fizz = i % divisor1 == 0;
            boolean buzz = i % divisor2 == 0;
            if (fizz && buzz) {
                out.append("FB ");
            } else if (fizz) {
                out.append("F ");
            } else if (buzz) {
                out.append("B ");
            } else {
                out.append(i).append(" ");
            }
        }
        System.out.println(out.toString().trim());
    }
}
